Close #996: Bump to Go 1.18 (#1011)
paketo-buildpacks/pipeline-builder: README.md

uses: docker://ghcr.io/paketo-buildpacks/actions/yourkit-dependency:main
```

## Updating Go version

> Dependabot could suggest to bump the Go version in some places, but you should make sure that the Go version is the same across the different tools in this repository.

Reviewer: ```suggestion
When it is time to update the Go version in the project, you should make sure that the Go version is the same across the different tools in this repository.
```

I'm trying to recall, but I don't think Dependabot will suggest updating them. In the past, we've initiated this manually. Go supports the most two recent versions, so we try to update as the second version is rolling out of support. That way we let the latest version stabilize a bit.

Most importantly, the Go version must be bumped in [Octo](octo/octo.go#L33) and in the [actions Dockerfile](actions/Dockerfile#L1) at the same time.

> Also, it could be a good idea to sync the go version in [the different workflows](.github/workflows).

Reviewer: ```suggestion
The Go version is also in [the different workflows](.github/workflows), however, this will be updated automatically by `octo`. You can run `octo` locally or submit a PR, and when the PR is merged CI will run `octo` and in turn submit another PR that updates the Go version.
```
